src/Core/Framework/Adapter/Twig/TokenParser/SwMacroFunctionTokenParser.php

```diff
@@ -0,0 +1,118 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Adapter\Twig\TokenParser;
+
+use Shopware\Core\Framework\Adapter\Twig\Node\MacroOverrideNode;
+use Shopware\Core\Framework\Log\Package;
+use Twig\Error\SyntaxError;
+use Twig\Node\BodyNode;
+use Twig\Node\EmptyNode;
+use Twig\Node\Expression\ArrayExpression;
+use Twig\Node\Expression\ConstantExpression;
+use Twig\Node\Expression\Unary\NegUnary;
+use Twig\Node\Expression\Unary\PosUnary;
+use Twig\Node\Expression\Variable\LocalVariable;
+use Twig\Node\Node;
+use Twig\Token;
+use Twig\TokenParser\AbstractTokenParser;
+use Twig\TokenParser\MacroTokenParser;
+
+#[Package('framework')]
+/**
+ * @internal
+ *
+ * @see MacroTokenParser -> basically copied, we use our own Macro node,
+ * that returns the actual instance of returned value instead of the markup
+ *
+ * @codeCoverageIgnore - Covered by @see \Shopware\Tests\Integration\Core\Framework\Adapter\Twig\ReturnNodeTest
+ */
+class SwMacroFunctionTokenParser extends AbstractTokenParser
+{
+    public function parse(Token $token): Node
+    {
+        $lineno = $token->getLine();
+        $stream = $this->parser->getStream();
+        $name = $stream->expect(Token::NAME_TYPE)->getValue();
+        $arguments = $this->parseDefinition();
+
+        $stream->expect(Token::BLOCK_END_TYPE);
+        $this->parser->pushLocalScope();
+        $body = $this->parser->subparse([$this, 'decideBlockEnd'], true);
```

Comment:
is that callable syntax deprecated? :thinking: 

would you mind trying: 

```suggestion
        $body = $this->parser->subparse($this->decideBlockEnd(...), true);
```

Reply:
This class is directly copied from twig, IMHO it makes more sense to leave it that way, so on upcoming updates it is easier to see when twig change something what exactly changed, now modifying "their" code only adds complexity, that's why the original classes are linked in the doc comment of those